plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerRegistryCredentialsReader.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.client;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.user.server.dao.PreferenceDao;
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.dto.server.DtoFactory;
+import org.eclipse.che.plugin.docker.client.dto.AuthConfig;
+import org.eclipse.che.plugin.docker.client.dto.AuthConfigs;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Base64;
+import java.util.Map;
+
+/**
+ * Class for reading credentials for docker registries from user preferences.
+ *
+ * @author Mykola Morhun
+ */
+@Singleton
+public class DockerRegistryCredentialsReader {
```

Comment:
Don't see the difference in names of this class and DockerRegistryAuthResolver. Rename to differentiate and show purpose of the class